### ISSUE TITLE
Semantics of CallSubOrchestrator in SDK is different from Core

### ISSUE THREAD

**Issue by michaelplavnik** [microsoft/durabletask-dotnet#187]
In error handling tests (e.g. NotFoundErrorsAreNotRetried) calling Bogus sub-orchestrator results in failure. In Core the same results in not completed task and increasing backoff of the dispatcher (questionable to say the least given that no timeout or cancellation option is provided). 

**Was this change of semantics intentional?**

**Comment by davidmrdavid** [comment]
@jviau can you comment on this?

**Comment by jviau** [comment]
> Was this change of semantics intentional?

@michaelplavnik do you see this behavior in Durable functions or standalone?

**Comment by michaelplavnik** [comment]
@jviau Standalone (worker)

**Comment by jviau** [comment]
Yeah, I see the logic for that, it looks intentional. We will need to discuss internally how we want to proceed here as it is a change from DurableTask.Core. Unfortunately, we won't be changing it until a major version rev (if at all).
